### PR TITLE
remove records of deleted pods from failure stores automatically and add an e2e test

### LINKED ISSUE #1464: [Advanced StatefulSet] e2e case: delete the failed pod after a failover
## Feature Request

**Describe the feature you'd like:**

an example:

- (normal) tc.spec.tikv.replicas = 3, sts.spec.replicas = 3
- (failover) tc.spec.tikv.replicas = 3, sts.spec.replicas = 4 ( tc.spec.tikv.replicas + len(tc.status.tikv.failureStores))
- delete the failed pod
  - if the failed pod is not expected to exist, we should delete it from the `cc.status.tikv.failureMembers`
- (after the deletion) tc.spec.tikv.replicas = 3, sts.spec.replicas = 3 ( tc.spec.tikv.replicas + len(tc.status.tikv.failureMembers))

### LINKED ISSUE #2459: tombstone stores are shown in the failurestores
## Bug Report

I deployed a TiDB cluster with TiFlash enabled on AWS with self-managed k8s cluster, each TiKV node has a tikv and tiflash pod. And then I've deleted one TiKV node, so tikv-1 and tiflash-0 becomes pending state. A new AWS ec2 instance was added after 10min. And then after 35min new tikv and tiflash was added and scheduled on the new ec2 instance.

Since I've used the advanced statefulset, by following the failover operation documentation, I've added the following annotations to `TidbCluster` object:

```yaml
annotations:
  tiflash.tidb.pingcap.com/delete-slots: '[0]'
  tikv.tidb.pingcap.com/delete-slots: '[1]'
```

The pending tikv was successfully removed, but tiflash was still there and the annotation was not propagated to tiflash AdvancedStatefulSet.

I also noted that the tombstone stores are in failurestore status:
image

### PR DESCRIPTION
**Pull request by cofyc**
### What problem does this PR solve? 
fixes #2459 
fixes #1464 
### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

**Comment by cofyc** [comment]
https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb-operator-e2e-kind-ci-master-stability/detail/tidb-operator-e2e-kind-ci-master-stability/88/pipeline

File: pkg/manager/member/tikv_member_manager.go [hunk 1]
```diff
@@ -201,6 +201,11 @@ func (tkmm *tikvMemberManager) syncStatefulSetForTidbCluster(tc *v1alpha1.TidbCl
 		return err
 	}
 
+	// Recover failed stores if any before generating desired statefulset
```
Comment (DanielZhangQD) [inline pkg/manager/member/tikv_member_manager.go:204]: We should also try to recover for TiFlash.

Comment (cofyc) [reply pkg/manager/member/tikv_member_manager.go:204]: done

**Pull request merged**